### PR TITLE
build(meson): add support for NetCDF in meson build system

### PR DESCRIPTION
**Pull request by jdhughes-usgs**
* rename PROFESSIONAL to EXTENDED version
* add function to test if __WITH_NETCDF__ define
* refactor is_pro to is_extended function that evaluates if either WITH_PETSC or WITH_NETCDF are defined
* add -Dextended=true meson option that will eventually replace -Dparallel=true

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ dependencies:
   - appdirs
   - filelock
   - fprettify
+  - fortran-language-server
```
Comment (wpbonelli) [inline environment.yml:12]: [this](https://github.com/fortran-lang/fortls#fortls-vs-fortran-language-server) makes it sound like `fortls` supersedes `fortran-language-server` but I have not used the latter so not sure.

Comment (jdhughes-usgs) [reply environment.yml:12]: The intent was to add the fortran-language-server to the development environment. Will change to fortls.

Comment (jdhughes-usgs) [reply environment.yml:12]: Actually it seems the modern fortran plugin for Visual Studio Code does not work with fortls but needs fortran-language-server (which has the same `fortls` exe name). See

https://stackoverflow.com/questions/72153608/fortran-language-server-fortls-error-in-vscode-windows

Comment (wpbonelli) [reply environment.yml:12]: Had a quick look at the fortls repo, it seems there are autodiscovery issues. Pointing vscode to it in workspace settings has worked if I recall, but that does not seem like it should be necessary and maybe there are other problems.

Sorry to lead astray, will look deeper in coming weeks

File: meson.build [hunk 2]
```diff
@@ -163,7 +166,23 @@ if is_parallel_build
...
   extra_cmp_args += [ '-D__WITH_MPI__']
   with_mpi = true
+
+  # find netcdf
+  if build_machine.system() != 'windows'
+    netcdf = dependency('netcdf', language : 'fortran', required : false)
+  else
+    # For CI testing only; Windows not yet supported
+    nc_dir = meson.project_source_root() / '..' / 'ncf' / 'netcdf-fortran-4.6.1' / 'fortran'
+    netcdf = fc.find_library('netcdff', dirs: [ nc_dir ], required : true, static : false)
```
Comment (mjreno) [inline meson.build:176]: required should be set false here to avoid a meson abort

**Review by mjreno — APPROVED** [review]

File: meson.build (hunk 2 above)
Comment (jdhughes-usgs) [reply meson.build]: eventually we need to decide if we are going to allow the `extended` version to cherry pick including petsc or netcdf

**Review by mjr-deltares — APPROVED** [review]

**Pull request merged**